src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -602,72 +891,131 @@ public void DeleteCountersForDocument(DocumentsOperationContext context, string
             if (table.NumberOfEntries == 0)
                 return;
 
-            using (GetCounterPartialKey(context, documentId, out var keyPrefix))
+            using (DocumentIdWorker.GetLowerIdSliceAndStorageKey(context, documentId, out Slice lowerId, out _))
             {
-                table.DeleteByPrimaryKeyPrefix(keyPrefix);
+                table.DeleteByPrimaryKeyPrefix(lowerId, holder => _countersCount -= GetData(context, ref holder.Reader).Count -1);
             }
         }
 
-        public string DeleteCounter(DocumentsOperationContext context, string documentId, string collection, string counterName)
+        public string DeleteCounter(DocumentsOperationContext context, Slice counterKey, string collection, long lastModifiedTicks, bool forceTombstone)
+        {
+            var (doc, name) = ExtractDocIdAndName(context, counterKey);
+            return DeleteCounter(context, doc, collection, name, forceTombstone, lastModifiedTicks);
+        }
+
+        public string DeleteCounter(DocumentsOperationContext context, string documentId, string collection, string counterName, bool forceTombstone = false, long lastModifiedTicks = -1)
```

Comment:
Do we _need_ a separate tombstone?
Can we mark the property itself on the object as deleted? 
Something like, if counter property is null, it is deleted?

Reply:
We can do that, but a couple of things to consider here:
- when deleting all counters in a group, we're left with a table entry that's all `null`s
- if a counter was deleted and then created again, how would we deal with that in replication? 
 probably need to keep a deleted etag to know which information is more up do date.

Reply:
Not sure, right now we are explicit that delete + modify on counter is something that user need to verify won't happen and might cause the modification to be lost or the delete to be lost

Reply:
removed CounterTombsones